source/jormungandr/jormungandr/scenarios/new_default.py

```diff
@@ -546,20 +546,26 @@ def _tag_by_mode(responses):
         for j in r.journeys:
             _tag_journey_by_mode(j)
 
-def _switch_back_to_ridesharing(response):
+
+def _is_fake_car(i, section, sections, dep_mode, arr_mode):
+    """
+    This function test if the section is a fake car section
+    """
+    return ((dep_mode == "ridesharing" and i == 0) or
+            (arr_mode == "ridesharing" and i == (len(sections)-1))) and \
+           ((section.type == response_pb2.STREET_NETWORK or section.type == response_pb2.CROW_FLY) and
+            section.street_network.mode == response_pb2.Car)
+
+
+def _switch_back_to_ridesharing(response, dep_mode, arr_mode):
     for journey in response.journeys:
         for i, section in enumerate(journey.sections):
-            if section.type == response_pb2.STREET_NETWORK \
-                    and section.street_network.mode == response_pb2.Car:
-                if (len(journey.sections) > i+1 and journey.sections[i+1].type != response_pb2.PARK) or \
-                        (i > 0 and journey.sections[i-1].type != response_pb2.LEAVE_PARKING) \
-                        or len(journey.sections) == 1:
-                    #TODO: handle crowfly :(
-                    section.street_network.mode = response_pb2.Ridesharing
-                    journey.durations.ridesharing += section.duration
-                    journey.durations.car -= section.duration
-                    journey.distances.ridesharing += section.length
-                    journey.distances.car -= section.length
+            if _is_fake_car(i, section, journey.sections, dep_mode, arr_mode):
```

Comment:
It looks a bit weird to loop over all sections when only the first and the last are of interest... To me a function for that is obfuscating the code for no gain (half of the params are useless depending on the case, if not all when on PT section).

Reply:
you're right, I hesitated A LOT and finally i left it that way :D, since you spot that, i'll change it